features/support/cucumber_spec_helper.rb

```diff
@@ -27,6 +27,11 @@ def cucumber_request_data data_type, doc
     if data_type.match /^(xml|json)$/i
       type = data_type.downcase.to_sym
       content_type = "application/#{type}"
+    else
+      matched_type = data_type.match /.*\+(xml|json)$/i
```

Comment:
Missing parentheses. Please also rewrite the regexp as `/(?:^application\/|.*\+)(xml|json)$/i` so that it matches `application/json` and `application/xml` as well.